engine/graphics/src/null/graphics_null_private.h

```diff
@@ -78,10 +79,33 @@ namespace dmGraphics
         FrameBuffer     m_FrameBuffer;
     };
 
-    struct Context
+    struct NullSharedAsset
+    {
+        enum AssetType
+        {
+            ASSET_TYPE_TEXTURE,
+            ASSET_TYPE_RENDER_TARGET,
+        };
+
+        NullSharedAsset()
+        {
+            memset(this, 0, sizeof(*this));
+        }
+
+        union
+        {
+            RenderTarget m_RenderTarget;
+            Texture      m_Texture;
+        };
+
+        AssetType m_Type;
+    };
```

Comment:
Right now we store all assets in the same container, so we carry around the type information with this shared struct. It's not really used anywhere, except for some validation here and there. This also means that the max number of shared assets is 65k since all of them will live in the same container.

A different approach that I'm thinking about is to represent the handle as a uint64_t instead and pack both the handle and the type into a "mixed handle-metadata" handle and store the handles in separate containers based on type. The benefits of this is that we don't need this extra indirection and don't need to mix handles in the same container. Should we allow exposing more assets like vertex buffers, shaders or uniform buffers in the future it might be nice to have this separation.. The drawback is of course that we "waste" the remaining bits in the handle, but maybe it's worth it?